src/main/java/seedu/address/logic/commands/FilterBuyersCommand.java

```diff
@@ -2,33 +2,42 @@
 
 import static java.util.Objects.requireNonNull;
 import static seedu.address.logic.parser.CliSyntax.PREFIX_CHARACTERISTICS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_MATCH_ALL;
 import static seedu.address.logic.parser.CliSyntax.PREFIX_PRICE;
 import static seedu.address.logic.parser.CliSyntax.PREFIX_PRIORITY;
 
+import java.util.function.Predicate;
+
 import seedu.address.commons.core.Messages;
 import seedu.address.model.Model;
-import seedu.address.model.buyer.AbstractFilterBuyerPredicate;
+import seedu.address.model.buyer.Buyer;
+
 
 /**
- * Filters and lists all buyers in the buyer list that either have a price range that accepts the given price,
- * characteristics that match the given characteristics list, or have the given tag.
+ * Filters and lists all buyers in the buyer list that have a price range that accepts the given price,
+ * contains all the characteristics in the given characteristics list, or have the given the priority.
+ * More than one filtering criteria can be accepted, buyers matching any or all of the given criteria can be returned
+ * based on whether the user passes in the 'PREFIX_MATCH_ALL' flag.
  * Keyword matching is case-insensitive.
  */
 public class FilterBuyersCommand extends Command {
 
     public static final String COMMAND_WORD = "filterbuyers";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all buyers in the database who"
-            + " either have a price range that accepts the given price, characteristics that match the given"
-            + " characteristics list, or has the given tag. You must filter by exactly one criteria at a time.\n"
+            + " either has a price range that accepts the given price, contains all characteristics in the given"
+            + " characteristics list, or has the given priority. You can pass in more than one criteria at a time.\n"
+            + " Pass in " + PREFIX_MATCH_ALL + " for strict filtering, i.e. filtered buyers will satisfy"
+            + " all the conditions specified. \n"
             + "Parameters: [" + PREFIX_PRICE + " PRICE] "
             + "[" + PREFIX_CHARACTERISTICS + " CHARACTERISTICS] "
-            + "[" + PREFIX_PRIORITY + " <HIGH/LOW>]\n"
+            + "[" + PREFIX_PRIORITY + " <HIGH/LOW>]"
+            + "[" + PREFIX_MATCH_ALL + "]\n"
             + "Example: " + COMMAND_WORD + " -c bright; sunny";
 
-    private final AbstractFilterBuyerPredicate predicate;
+    private final Predicate<Buyer> predicate;
 
-    public FilterBuyersCommand(AbstractFilterBuyerPredicate predicate) {
+    public FilterBuyersCommand(Predicate<Buyer> predicate) {
```

Comment:
oh this is because taking `Predicate::or` or `Predicate::and` returns a `Predicate<Buyer>` and not the abstract class which causes some type errors

Reply:
Yes I believe this is the case. In v1.4 we should probably remove the abstract classes for both buyer and property predicates. Not sure what's the point of their existence since they only override a method from `Predicate<T>` without implementing any new ones.